clients/fides-js/src/components/tcf/TcfVendors.tsx

```diff
@@ -226,34 +281,18 @@ const TcfVendors = ({
   const [activeLegalBasisOption, setActiveLegalBasisOption] = useState(
     LEGAL_BASIS_OPTIONS[0]
   );
-  const activeData: {
```

Comment:
This is mostly moving the contents of the vendor data to a child component. This allows us to think of the toggles ("consent" vs "legitimate interest") as top level filters, which inform what data gets shown in the child component. We need either the legal basis filter or paging to be top level (we page through _by_ legal basis)